code/game/objects/items/weapons/traps.dm

```diff
@@ -656,3 +641,44 @@
 		return
 	else
 		..()
+
+
+/obj/item/trap/tripwire
+	name = "tripwire trap"
+	icon_base = "tripwire"
+	color = COLOR_RED
+	layer = UNDERDOOR // so you can't cover it with items
+
+/obj/item/trap/tripwire/Initialize(mapload, var/new_cable_color)
+	. = ..()
+	if(!new_cable_color)
+		new_cable_color = COLOR_RED
+	color = new_cable_color
+
+/obj/item/trap/tripwire/update_icon()
+	underlays = null
+	icon_state = "[icon_base][deployed]"
+	var/image/I = image(icon, null, "[icon_state]_base")
+	I.appearance_flags = RESET_COLOR
+	underlays = list(I)
+
+/obj/item/trap/tripwire/deploy(mob/user)
+	user.visible_message(SPAN_WARNING("\The [user] starts to deploy \the [src]."), SPAN_WARNING("You begin deploying \the [src]!"))
+	if(do_after(user, 5 SECONDS))
+		user.visible_message(SPAN_WARNING("\The [user] deploys \the [src]."), SPAN_WARNING("You deploy \the [src]!"))
+		deployed = TRUE
+		update_icon()
+		return TRUE
+	return FALSE
+
+/obj/item/trap/tripwire/disarm_trap(var/mob/user)
+	user.visible_message(SPAN_NOTICE("\The [user] starts to take \the [src] down..."), SPAN_NOTICE("You begin taking \the [src] down..."), SPAN_WARNING("You hear a latch click followed by the slow creaking of a spring."))
+	if(do_after(user, 6 SECONDS))
+		deployed = FALSE
+		anchored = FALSE
+		update_icon()
+
+/obj/item/trap/tripwire/attack_mob(mob/living/L)
```

Comment:
Wouldn't this trip mechs? Also doesn't take into account living mobs that fly/float. 

Additionally I'm pretty concerned about how easy it would be to spam a hallway of these with access to basic tools. Since it's a makeshift trap, I wonder if making it have a chance of triggering even on run intent would be better than a guaranteed thing.

Reply:
you need two sets of cable coil to make a single tripwire trap, which makes spamming it impossible without a LOT of effort, with the same effort you can make a gun that does actual damage instead.

i've updated the trap to only check for humans and whether they have legs, i assume stuff without legs fly and wouldn't trip over the trap. as for the chance to trip while running, i don't really like the idea of your perfect trap not working in the absolute ideal conditions, but we'll need to see how it plays out ingame

Reply:
I misread and didn't see it used the entire stack of coil, so that alleviates my fears of spamming it. And without it being spammed, the surefire thing seems fine to try out.